docs/_static/didman/v1.yaml

```diff
@@ -188,6 +188,59 @@ paths:
                 $ref: '#/components/schemas/CompoundService'
         default:
           $ref: '../common/error_response.yaml'
+  /internal/didman/v1/did/{did}/compoundservice/{compoundServiceType}/endpoint/{endpointType}:
+    parameters:
+      - name: did
+        in: path
+        description: URL encoded DID.
+        required: true
+        example:
+          - "did:nuts:1234"
+        schema:
+          type: string
+      - name: compoundServiceType
+        in: path
+        description: Service type of the compound service containing the endpoint to be resolved.
+        required: true
+        example: geolocation
+        schema:
+          type: string
+      - name: endpointType
+        in: path
+        description: Entry in the compound service to be resolved as endpoint.
+        required: true
+        example: geolocation-rest-api
+        schema:
+          type: string
+    get:
+      operationId: getCompoundServiceEndpoint
+      summary: |
+        Retrieves the endpoint with the specified endpointType from the specified compound service.
+        It returns the serviceEndpoint of the specified service (which must be an absolute URL endpoint).
+
+        error responses:
+        * 400 - incorrect input (e.g. the given service type isn't a compound service)
+        * 404 - unknown DID, compound service or endpoint
+        * 406 - service references are nested too deep or reference is invalid in other ways
```

Comment:
maybe a 400 is better. 406 is returned based on the accept headers. The underlying error is only returned when `resolve == true`, when it is `false` it'll succeed. Hence bad query input??

Reply:
It can also mean the service isn't correct (nested too deep), but which is done by the other party. 400 bad request implies (for me) that you need to alter your input for your request to succeed. But it's the data (entered by the remote party) that corrupt, so you won't be able to make the call succeed by altering your input.

Hence the status code != 400?

Reply:
There's really nothing that naturally fits, and it also doesn't matter too much. As long as the returned code isn't confusing. I associate status 500 with something that's gone terribly bad in the internals of the server, which isn't actually the case either.

Maybe 422 Unprocessable Entity?

> The HyperText Transfer Protocol (HTTP) 422 Unprocessable Entity response status code indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.

Reply:
one of the most vague codes and therefore a good match? ;)